src/rules/prefer-importing-jest-globals.ts

```diff
@@ -0,0 +1,186 @@
+import type { Literal } from 'estree';
+import { type ParsedJestFnCall, createRule, parseJestFnCall } from './utils';
+
+const createFixerImports = (
+  usesImport: boolean,
+  functionsToImport: string[],
+) => {
+  const allImportsFormatted = functionsToImport.filter(Boolean).join(', ');
+
+  return usesImport
+    ? `import { ${allImportsFormatted} } from '@jest/globals';`
+    : `const { ${allImportsFormatted} } = require('@jest/globals');`;
+};
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Prefer importing Jest globals',
+      recommended: false,
+    },
+    messages: {
+      preferImportingJestGlobal: `Import the following Jest functions from '@jest/globals': {{ jestFunctions }}`,
+    },
+    fixable: 'code',
+    type: 'problem',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const importedJestFunctions: string[] = [];
+    const usedJestFunctions: ParsedJestFnCall[] = [];
+
+    return {
+      CallExpression(node) {
+        const jestFnCall = parseJestFnCall(node, context);
+
+        if (!jestFnCall) {
+          return;
+        }
+
+        if (jestFnCall.head.type === 'import') {
+          importedJestFunctions.push(jestFnCall.name);
+        }
+
+        usedJestFunctions.push(jestFnCall);
```

Comment:
if there are no existing `require` statements, it should be suggestions of either format so the user selects explicitly

Reply:
I wonder if ESLint exposes the `sourceType` option to us through `context`? that'd be a reasonable way of deciding automatically which to use right?

Reply:
I can use the `context.parserOptions.sourceType` to decide automatically the right fix 👍

Reply:
oooh, nice! didn't think about that one 👍